folding-schemes/src/folding/nova/circuits.rs

```diff
@@ -391,6 +406,11 @@ where
         )?;
         let r = Boolean::le_bits_to_fp_var(&r_bits)?;
 
+        // enforce that the input r_nonnat as bits matches the in-circuit computed r_bits
+        let r_nonnat_bits: Vec<Boolean<C1::ScalarField>> =
+            r_nonnat.to_bits_le()?.into_iter().take(128).collect();
```

Comment:
Oh this is a mistake from my side! this should come from the global constant `N_BITS_RO`. The reason for it being 128 comes from [Srinath Setty](https://microsoft.com/en-us/research/people/srinath/): *"In Nova, soundness error ≤ 2/|S|, where S is the subset of the field F from which the challenges are drawn. In this case, we keep the size of S close to 2^128."* (documented in the [constant](https://github.com/privacy-scaling-explorations/folding-schemes/blob/main/folding-schemes/src/constants.rs#L2)).
[Updated](https://github.com/privacy-scaling-explorations/folding-schemes/compare/872bdcc2706b2f57a0b62d13466c65b199c75d01..a401af0adf8a6e7b23292b7ee92fe838fc289cbb) this PR to use the constant instead of the hardcoded value.
Thanks for pointing it out!